### PR TITLE
chore: add integration tests for AppBundle, AppUpload and AppDefinition

### PR DESCRIPTION
**Pull request by FLoppix**
## Summary

Adding basic integration integration tests for the following entities:

- `AppDefinition`
- `AppUpload`
- `AppBundle`

## Motivation and Context

Improving test coverage

## Checklist (check all before merging)

- [x] Both unit and integration tests are passing
- [x] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [ ] The new method is exported through the default and plain CMA client
- [ ] All new public types are exported from `./lib/export-types.ts`
- [ ] Added a unit test for the new method
- [x] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

File: test/integration/app-definition-integration.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          expect(response.sys.id).equals(newAppDefinition.sys.id, 'id')
+          expect(response.name).equals('Test App Updated', 'name')
+
+          return organization
+            .getAppDefinitions()
+            .then((response) => {
+              expect(response.items.length).equals(
+                response.total,
+                'returns the just created app definitions'
+              )
```
Comment (andipaetzold) [inline test/integration/app-definition-integration.js:51]: Isn't this just a test of whether the pagination works? Feels quite unspecific.

File: test/integration/app-definition-integration.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  before(async () => {
+    organization = await client()
+      .getOrganizations()
+      .then((response) => response.items[0])
+  })
+
+  after(async () => {
+    organization.getAppDefinitions().then((response) => {
+      response.items.map((appDefinition) => appDefinition.delete())
+    })
```
Comment (andipaetzold) [inline test/integration/app-definition-integration.js:17]: You need to use `await Promise.all(...)` here. Or do this:

```suggestion
    const appDefinitions = await organization.getAppDefinitions();
    for (const appDefinition of appDefinitions) {
      await appDefinition.delete()
    }
```

File: test/integration/app-bundle-integration.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  after(async () => {
+    organization.getAppDefinitions().then((response) => {
+      response.items.map((appDefinition) => appDefinition.delete())
+    })
+
+    if (appUpload) {
+      appUpload.delete()
+    }
+  })
```
Comment (andipaetzold) [inline test/integration/app-bundle-integration.js:31]: Same as in `app-definition-integration`

File: test/integration/app-bundle-integration.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  test('Create, get, get all and delete AppBundle', async () => {
+    return appDefinition
+      .createAppBundle({ appUploadId: appUpload.sys.id, comment: 'Test comment' })
+      .then((appBundle) => {
+        expect(appBundle.sys.type).equals('AppBundle', 'type')
+        expect(appBundle.comment).equals('Test comment', 'comment')
+        expect(appBundle.files).to.be.an('array')
```
Comment (andipaetzold) [inline test/integration/app-bundle-integration.js:39]: Should we check for the array content to make sure it actually contains something?

File: test/integration/app-bundle-integration.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        return appDefinition.getAppBundle(appBundle.sys.id).then((response) => {
+          expect(response.sys.id).equals(appBundle.sys.id)
+
+          return appDefinition
+            .getAppBundles()
+            .then((response) => {
+              expect(response.items.length).equals(
+                response.total,
+                'returns the just created app bundles'
+              )
```
Comment (andipaetzold) [inline test/integration/app-bundle-integration.js:49]: I don't get this test

**Review by andipaetzold — CHANGES_REQUESTED** [review]

File: test/integration/app-bundle-integration.js (hunk 4 above)
Comment (FLoppix) [reply test/integration/app-bundle-integration.js:39]: Yep will add a check to see if we upload an `index.html` file !

File: test/integration/app-bundle-integration.js [hunk 6]
```diff
@@ -21,35 +21,61 @@ describe('AppBundle api', function () {
...
+  test('createAppBundle', async () => {
+    const appBundle = await appDefinition.createAppBundle({
+      appUploadId: appUpload.sys.id,
+      comment: 'Test comment',
+    })
+
+    expect(appBundle.sys.type).equals('AppBundle', 'type')
+    expect(appBundle.comment).equals('Test comment', 'comment')
+    expect(appBundle.files).to.be.an('array')
+    const indexFile = appBundle.files.filter((file) => (file.name = 'build/index.html'))
```
Comment (andipaetzold) [inline test/integration/app-bundle-integration.js:43]:
```suggestion
    const indexFile = appBundle.files.filter((file) => (file.name === 'build/index.html'))
```

**Review by andipaetzold — CHANGES_REQUESTED** [review]
Looks good. Just one error that makes the test always succeed

**Review by andipaetzold — APPROVED** [review]

**Pull request merged**